components/plan-build/bin/hab-plan-build.sh

```diff
@@ -1598,13 +1604,16 @@ _build_environment() {
   done
   for dep_path in "${pkg_deps_resolved[@]}"; do
     if [[ -f "$dep_path/LD_RUN_PATH" ]]; then
-      local data=$(cat $dep_path/LD_RUN_PATH)
-      local trimmed=$(trim $data)
+      local data
+      data=$(cat "$dep_path"/LD_RUN_PATH)
+      local trimmed
+      trimmed=$(trim "$data")
       ld_run_path_part+=("$trimmed")
     fi
   done
-  if [[ -n $ld_run_path_part ]]; then
-    export LD_RUN_PATH=$(join_by ':' ${ld_run_path_part[@]})
+  if [[ -n "${ld_run_path_part[*]}" ]]; then
```

Comment:
`ld_run_path_part` is an array, so it's best to consistently treat it like one. `$ld_run_path_part` would access just the first element, which will usually give the desired behavior in this condition, but there are pathological cases which break, such as:
```bash
an_array=("" hello)
```

In the end, perhaps it's preferable to eliminate the condition altogether and unconditionally execute
```bash
  export LD_RUN_PATH
  LD_RUN_PATH=$(join_by ':' "${ld_run_path_part[@]}")
```
Is there a downside to that?

Reply:
After loading [join_by](https://github.com/habitat-sh/habitat/blob/master/components/plan-build/bin/public.bash#L290-L294):
```bash
jugokai:habitat jbauman$ ld_run_path_part=()
jugokai:habitat jbauman$ LD_RUN_PATH=$(join_by ':' "${ld_run_path_part[@]}")
jugokai:habitat jbauman$ declare -p LD_RUN_PATH
declare -- LD_RUN_PATH=""
jugokai:habitat jbauman$ ld_run_path_part=("" foo)
jugokai:habitat jbauman$ LD_RUN_PATH=$(join_by ':' "${ld_run_path_part[@]}")
jugokai:habitat jbauman$ declare -p LD_RUN_PATH
declare -- LD_RUN_PATH=":foo"
```

[`declare -p`](https://www.gnu.org/software/bash/manual/html_node/Bash-Builtins.html#index-declare) is a handy way of inspecting a variable:
```bash
jugokai:habitat jbauman$ some_empty_string=""
jugokai:habitat jbauman$ echo _$some_empty_string_
jugokai:habitat jbauman$ echo x${some_empty_string}x
xx
jugokai:habitat jbauman$ echo x${ld_run_path_part}x
xx
jugokai:habitat jbauman$ declare -p some_empty_string
declare -- some_empty_string=""
jugokai:habitat jbauman$ declare -p ld_run_path_part
declare -a ld_run_path_part=([0]="" [1]="foo")
```